supervisor/backups/backup.py

```diff
@@ -515,16 +477,32 @@ async def restore_homeassistant_config_dir(self):
 
         await self.sys_homeassistant.restore(homeassistant_file)
 
+        # Generate restore task
+        async def _core_update():
+            try:
+                if self.homeassistant_version == self.sys_homeassistant.version:
+                    return
+            except (AwesomeVersionCompareException, TypeError):
+                pass
+            await self.sys_homeassistant.core.update(self.homeassistant_version)
```

Comment:
Since we get here on both AwesomeVersionCompareException and TypeError, we can now pass an invalid version to `core.update(...`
Should we at least verify `self.homeassistant_version`?